src/relayer/Relayer.ts

```diff
@@ -36,141 +37,154 @@ export class Relayer {
   }
 
   /**
-   * @description Retrieve the complete array of unfilled deposits and filter out deposits we can't or choose
-   * not to support.
-   * @returns An array of filtered RelayerUnfilledDeposit objects.
+   * @description For a given deposit, apply relayer-specific filtering to determine whether it should be filled.
+   * @param deposit Deposit object.
+   * @param version Version identified for this deposit.
+   * @param invalidFills An array of any invalid fills detected for this deposit.
+   * @returns A boolean indicator determining whether the relayer configuration permits the deposit to be filled.
    */
-  private async _getUnfilledDeposits(): Promise<RelayerUnfilledDeposit[]> {
-    const { configStoreClient, hubPoolClient, spokePoolClients, acrossApiClient } = this.clients;
-    const { relayerTokens, ignoredAddresses, acceptInvalidFills } = this.config;
-
-    // Flatten unfilledDeposits for now. @todo: Process deposits in parallel by destination chain.
-    const unfilledDeposits = Object.values(
-      await getUnfilledDeposits(spokePoolClients, hubPoolClient, this.config.maxRelayerLookBack)
-    ).flat();
-
-    const maxVersion = configStoreClient.configStoreVersion;
-    return sdkUtils.filterAsync(unfilledDeposits, async ({ deposit, version, invalidFills }) => {
-      const { depositId, depositor, recipient, originChainId, destinationChainId, inputToken, outputToken } = deposit;
-      const destinationChain = getNetworkName(destinationChainId);
-
-      // If we don't have the latest code to support this deposit, skip it.
-      if (version > maxVersion) {
-        this.logger.warn({
-          at: "Relayer::getUnfilledDeposits",
-          message: "Skipping deposit that is not supported by this relayer version.",
-          latestVersionSupported: maxVersion,
-          latestInConfigStore: configStoreClient.getConfigStoreVersionForTimestamp(),
-          deposit,
-        });
-        return false;
-      }
+  filterDeposit({ deposit, version, invalidFills }: RelayerUnfilledDeposit): boolean {
+    const { depositId, originChainId, destinationChainId, depositor, recipient, inputToken, outputToken } = deposit;
+    const { acrossApiClient, configStoreClient, hubPoolClient } = this.clients;
+    const { ignoredAddresses, relayerTokens, acceptInvalidFills } = this.config;
 
-      if (!this.routeEnabled(originChainId, destinationChainId)) {
-        this.logger.debug({
-          at: "Relayer::getUnfilledDeposits",
-          message: "Skipping deposit from or to disabled chains.",
-          deposit,
-          enabledOriginChains: this.config.relayerOriginChains,
-          enabledDestinationChains: this.config.relayerDestinationChains,
-        });
-        return false;
-      }
+    // If we don't have the latest code to support this deposit, skip it.
+    if (version > configStoreClient.configStoreVersion) {
```

Comment:
```suggestion
    if (configStoreClient.configStoreVersion < version) {
```

Nit: could we flip the comparison? They're functionally the same but IMO this is easier to read

Reply:
Resolved out of band: https://github.com/across-protocol/relayer-v2/pull/1346/commits/451b67f15b4982c0f54afdf42d5525d65e9eeed4-